File: apache_spark/4_word_count.py
from pyspark import SparkConf, SparkContext


conf = SparkConf().setMaster("local").setAppName("WordCount")
sc = SparkContext(conf = conf)
sc.setLogLevel('FATAL')


lines = sc.textFile("/Users/zhedongzheng/tutorials/apache_spark/temp/Book.txt")
words = lines.flatMap(lambda x: x.split())
wordCounts = words.countByValue()


for word, count in wordCounts.items():
    cleanWord = word.encode('ascii', 'ignore')
    if (cleanWord):
        print(cleanWord.decode() + " " + str(count))

File: apache_spark/4_word_count.scala
import org.apache.spark._
import org.apache.spark.SparkContext._
import org.apache.log4j._

   
// Set the log level to only print errors
Logger.getLogger("org").setLevel(Level.ERROR)

  // Create a SparkContext using every core of the local machine
val sc = new SparkContext("local[*]", "WordCount")   

// Read each line of my book into an RDD
val lines = sc.textFile("/Users/zhedongzheng/tutorials/apache_spark/temp/Book.txt")

// Split into words separated by a space character
val words = lines.flatMap(x => x.split(" "))

// Count up the occurrences of each word
val wordCounts = words.countByValue()

// Print the results.
wordCounts.foreach(println)